src/sections/reports.ts

```diff
@@ -0,0 +1,623 @@
+import {
+  array,
+  boolean,
+  Codec,
+  enumeration,
+  exactly,
+  GetInterface,
+  Left,
+  number,
+  oneOf,
+  optional,
+  string,
+} from 'purify-ts'
+
+import { ParsingError } from '../error'
+import { HttpClient, RequestMeta, Resource } from '../http'
+import { ensureString, mwsDate, NextToken, nextToken as nextTokenCodec } from '../parsing'
+
+const REPORTS_API_VERSION = '2009-01-01'
+/**
+ * List of supported strings.
+ * Could probably define an enum for this
+ * http://docs.developer.amazonservices.com/en_CA/reports/Reports_ReportType.html#ReportTypeCategories__ListingsReports
+ */
+const ReportType = string
+type ReportType = GetInterface<typeof ReportType>
+
+interface RequestReportParameters {
+  ReportType: ReportType
+  StartDate?: Date
+  EndDate?: Date
+  ReportOptions?: string
+  MarketplaceIdList?: string[]
+}
+
+// Amazon does not explicitlye define which elements of the response are required so everything is optional for now
```

Comment:
I forgot, I should've mentioned this earlier. This was a tricky slope when I came across it, because if I "require" the wrong thing it'll throw a parsing error and from what I see in the docs, anything could be optional. Like in `requestReport`'s case you'd expect `GeneratedReportId` to be required in the response but it's actually not, but it's in the docs